trace-utils/src/send_data.rs

```diff
@@ -383,17 +517,410 @@ impl SendData {
     use tokio::time::Instant;
 
     const RETRY_STRATEGY_TIME_TOLERANCE_MS: u64 = 25;
+    const HEADER_TAGS: TracerHeaderTags = TracerHeaderTags {
+        lang: "test-lang",
+        lang_version: "2.0",
+        lang_interpreter: "interpreter",
+        lang_vendor: "vendor",
+        tracer_version: "1.0",
+        container_id: "id",
+        client_computed_top_level: false,
+        client_computed_stats: false,
+    };
+
+    fn setup_payload(header_tags: &TracerHeaderTags) -> pb::TracerPayload {
+        let root_tags = RootSpanTags {
+            env: "TEST",
+            app_version: "1.0",
+            hostname: "test_bench",
+            runtime_id: "id",
+        };
+
+        let chunk = construct_trace_chunk(vec![pb::Span {
+            service: "test-service".to_string(),
+            name: "test-service-name".to_string(),
+            resource: "test-service-resource".to_string(),
+            trace_id: 111,
+            span_id: 222,
+            parent_id: 333,
+            start: 1,
+            duration: 5,
+            error: 0,
+            meta: HashMap::new(),
+            metrics: HashMap::new(),
+            meta_struct: HashMap::new(),
+            r#type: "".to_string(),
+            span_links: vec![],
+        }]);
+
+        construct_tracer_payload(vec![chunk], header_tags, root_tags)
+    }
+
+    fn compute_payload_len(payload: &[pb::TracerPayload]) -> usize {
+        let agent_payload = construct_agent_payload(payload.to_vec());
+        let serialized_trace_payload = serialize_proto_payload(&agent_payload).unwrap();
+        serialized_trace_payload.len()
+    }
+    fn rmp_compute_payload_len(payload: &Vec<pb::TracerPayload>) -> usize {
+        let mut total: usize = 0;
+        for payload in payload {
+            total += rmp_serde::to_vec_named(payload).unwrap().len();
+        }
+        total
+    }
+
+    #[test]
+    fn send_data_new_api_key() {
+        let header_tags = TracerHeaderTags {
+            lang: "test-lang",
+            lang_version: "2.0",
+            lang_interpreter: "interpreter",
+            lang_vendor: "vendor",
+            tracer_version: "1.0",
+            container_id: "id",
+            client_computed_top_level: false,
+            client_computed_stats: false,
+        };
+
+        let payload = setup_payload(&header_tags);
+        let data = SendData::new(
+            100,
+            payload,
+            HEADER_TAGS,
+            &Endpoint {
+                api_key: Some(std::borrow::Cow::Borrowed("TEST-KEY")),
+                url: "/foo/bar?baz".parse::<hyper::Uri>().unwrap(),
+            },
+        );
+
+        assert_eq!(data.size, 100);
+
+        assert_eq!(data.target.api_key.unwrap(), "TEST-KEY");
+        assert_eq!(data.target.url.path(), "/foo/bar");
+
+        assert_eq!(data.headers.get("DD-API-KEY").unwrap(), "TEST-KEY");
+    }
+
+    #[test]
+    fn send_data_new_no_api_key() {
+        let header_tags = TracerHeaderTags {
```

Comment:
you could probably just create a `get_tracer_header_tags()` function or implement a default trait and add a `#[cfg(test)]` to it to avoid having to create one in every test.

Reply:
Good catch, I didn't realize I was cluttering the tests when I could use the default impl.